assignment1/string_permutation.py

```diff
@@ -0,0 +1,18 @@
+import string 
+
+def string_permutation(str1, str2):
+    if (str.isalpha(str1) or str1 == "") and (
+            str.isalpha(str2) or str2 == ""):
+        str1 = str1.lower()
+        str2 = str2.lower()
+        inputs = {x: 0 for x in string.ascii_lowercase} 
```

Comment:
This is fine but just as an FYI - `setdefault(...)` and `collections.defaultdict(...)` might also be considered here:

```
inputs = collections.defaultdict(lambda: 0)
for c in str1:
  inputs[c] += 1
...
```